php_phongo.c

```diff
@@ -1880,11 +1878,17 @@ PHP_MINIT_FUNCTION(mongodb)
 {
 	(void)type; /* We don't care if we are loaded via dl() or extension= */
 
+	char *php_version_string = malloc(4 + sizeof(MONGODB_VERSION_S) + 1);
```

Comment:
`old_str` in the code that you linked is the libmongoc global. The string you're creating here goes into that function as a `const char *` as well under the `suffix` param (see: [code](https://github.com/mongodb/mongo-c-driver/blob/8fd25223acba9e63fe81c1d4712890cad96163b8/src/mongoc/mongoc-handshake.c#L430)).

If I had to guess, `emalloc()` segfaults because you're creating this outside of a request. I recall reading something about persistent allocations using `malloc()` instead of `emalloc()` in the PHP Internals Book ([here](http://www.phpinternalsbook.com/zvals/memory_management.html#allocating-and-initializing-zvals)). In this case, I think you should still `free()` after you append the value.